### PR TITLE
Modify find command

### PR DESCRIPTION
**Pull request by PateShin**

### CODE REVIEW

File: src/main/java/seedu/address/model/person/TagEqualsPredicate.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    public TagEqualsPredicate(String tagName) {
+        this.tagName = tagName;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return person.getTags().stream()
+                .anyMatch(tag -> tag.tagName.equals(tagName));
+    }
```
Comment (linnn-7) [inline src/main/java/seedu/address/model/person/TagEqualsPredicate.java:19]: Maybe can use .map(x -> x.toLowerCase()) to set all tags to lowercase to compare their value.

File: src/main/java/seedu/address/model/person/TagEqualsPredicate.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Tests that a {@code Person} has a tag that matches the given tag name.
+ */
+public class TagEqualsPredicate implements Predicate<Person> {
+    private final String tagName;
+
+    public TagEqualsPredicate(String tagName) {
+        this.tagName = tagName;
+    }
+
```
Comment (linnn-7) [inline src/main/java/seedu/address/model/person/TagEqualsPredicate.java:14]: Maybe can use .toLowerCase() to set it to lowercase.

File: src/test/java/seedu/address/logic/parser/FindCommandParserTest.java [hunk 3]
```diff
@@ -1,34 +1,93 @@
...
+    public void parse_validNameArgs_returnsFindCommand() {
+        // no leading and trailing whitespaces
+        FindCommand expectedFindCommand =
+                new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList("Alice", "Bob")));
+        assertParseSuccess(parser, "/find :name Alice Bob", expectedFindCommand);
+
+        // multiple whitespaces between keywords
+        assertParseSuccess(parser, "/find :name Alice      Bob  ", expectedFindCommand);
+    }
+
```
Comment (linnn-7) [inline src/test/java/seedu/address/logic/parser/FindCommandParserTest.java:39]: I am grateful for your efforts in incorporating additional test cases to ensure broader coverage of our program's code.

**Review by linnn-7 — COMMENTED** [review]
Overall, your code is clear and efficiently accomplishes tasks in a concise manner. LGTM!

File: src/main/java/seedu/address/model/person/TagEqualsPredicate.java (hunk 2 above)
Comment (PateShin) [reply src/main/java/seedu/address/model/person/TagEqualsPredicate.java]: Thanks for pointing I have added the feature to existing code.

File: src/main/java/seedu/address/model/person/TagEqualsPredicate.java (hunk 1 above)
Comment (PateShin) [reply src/main/java/seedu/address/model/person/TagEqualsPredicate.java]: Thanks for your feedback! I have modified the code with feedback that you suggested.

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 4]
```diff
@@ -55,4 +57,8 @@ public String toString() {
                 .add("predicate", predicate)
                 .toString();
     }
+
+    public Predicate<Person> getPredicate() {
+        return predicate;
```
[the remark this answers is not in the document]
Comment (PateShin) [reply src/main/java/seedu/address/logic/commands/FindCommand.java:62]: The `@Test` is added to `FindCommandTest.java` to test `getPredicate()`.

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 5]
```diff
@@ -19,15 +25,30 @@ public class FindCommandParser implements Parser<FindCommand> {
...
+            return new FindCommand(new IdEqualsPredicate(
+                    argMultimap.getValue(CliSyntax.PREFIX_ID).get()));
+        } else if (argMultimap.getValue(CliSyntax.PREFIX_YEAR_JOINED).isPresent()) {
+            return new FindCommand(new YearJoinedEqualsPredicate(
+                    argMultimap.getValue(CliSyntax.PREFIX_YEAR_JOINED).get()));
+        } else if (argMultimap.getValue(CliSyntax.PREFIX_TAG).isPresent()) {
+            return new FindCommand(new TagEqualsPredicate(
+                    argMultimap.getValue(CliSyntax.PREFIX_TAG).get()));
+        } else {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
```
Comment (ondretann) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:51]: This method seems too long, you can consider abstracting it out to another method for better readability.

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 6]
```diff
@@ -188,9 +171,13 @@ public void parse_invalidValue_failure() {
...
 
+        // invalid tag
+        assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + YEARJOINED_DESC_BOB
+                + ADDRESS_DESC_BOB + INVALID_TAG_DESC + VALID_TAG_FRIEND, Tag.MESSAGE_CONSTRAINTS);
+
         // two invalid values, only first invalid value reported
-        assertParseFailure(parser, INVALID_NAME_DESC + PHONE_DESC_BOB + EMAIL_DESC_BOB + INVALID_ADDRESS_DESC,
-                Name.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + YEARJOINED_DESC_BOB
+                + ADDRESS_DESC_BOB + INVALID_TAG_DESC + VALID_TAG_FRIEND, Tag.MESSAGE_CONSTRAINTS);
```
Comment (ondretann) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:180]: Test cases are very thorough and comprehensive. Nice!

**Pull request merged**